[upb]BUILD_FAILED on x64-windows

Building the latest version of QT today (July 11th), I got this build error message
```
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: C:/Tools/cmake/bin/cmake.exe --build . --config Debug --target install -- -v -j25
    Working Directory: E:/Dev/vcpkg/buildtrees/upb/x64-windows-dbg
    See logs for more information:
      E:\Dev\vcpkg\buildtrees\upb\install-x64-windows-dbg-out.log
```


The error inside the file '**install-x64-windows-dbg-out.log**' is:
`E:\Dev\vcpkg\buildtrees\upb\src\v4.25.1-328a82a975.clean\upb\message\message.c(20): error C2099: initializer is not a constant
`
The line **20** of '**message.c**' is:
`const double kUpb_NaN = NAN;`

--- a/ports/upb/fix-NAN-on-Win11.patch
+++ b/ports/upb/fix-NAN-on-Win11.patch
@@ -1,0 +1,17 @@
+diff --git a/upb/message/message.c b/upb/message/message.c
+index 265d30d..afc6e51 100644
+--- a/upb/message/message.c
++++ b/upb/message/message.c
+@@ -15,9 +15,9 @@
+ // Must be last.
+ #include "upb/port/def.inc"
+ 
+-const float kUpb_FltInfinity = INFINITY;
+-const double kUpb_Infinity = INFINITY;
+-const double kUpb_NaN = NAN;
++const float kUpb_FltInfinity = (float)(1.0 / 0.0);
++const double kUpb_Infinity = 1.0 / 0.0;
++const double kUpb_NaN = 0.0 / 0.0;
+ 
+ static const size_t overhead = sizeof(upb_Message_InternalData);
+ 

--- a/ports/upb/portfile.cmake
+++ b/ports/upb/portfile.cmake
@@ -8,6 +8,7 @@ vcpkg_from_github(
     HEAD_REF main
     PATCHES
         fix-cmake.patch
+        fix-NAN-on-Win11.patch
 )
 
 vcpkg_check_features(

--- a/ports/upb/vcpkg.json
+++ b/ports/upb/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "upb",
   "version": "4.25.1",
+  "port-version": 1,
   "description": "μpb (often written 'upb') is a small protobuf implementation written in C.",
   "homepage": "https://github.com/protocolbuffers/protobuf",
   "license": "BSD-2-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -9054,7 +9054,7 @@
     },
     "upb": {
       "baseline": "4.25.1",
-      "port-version": 0
+      "port-version": 1
     },
     "urdfdom": {
       "baseline": "3.1.1",

--- a/versions/u-/upb.json
+++ b/versions/u-/upb.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "47d5e848e2c7213cc67e4dfa6c7797fa973f2c12",
+      "version": "4.25.1",
+      "port-version": 1
+    },
+    {
       "git-tree": "3edd5c1f37803c69faaf1b0884791e8c1d8ff48f",
       "version": "4.25.1",
       "port-version": 0

